### PR TITLE
Second batch of R tests

### PR DESCRIPTION
**Pull request by capnrefsmmat**
This addresses a few of the items in #239.

The httptest package will be very helpful here. I was going to use `mockery::stub`, but it misbehaves when you want to repeatedly swap out the stub to return different values. Instead, httptest serves the API HTTP requests from a file instead of letting them go to the server. See the comment at the top of `test-covidcast.R` for more details.

It's a bit clunky (particularly keeping track of the JSON) but it is *much* better than having no tests at all.

cc @JedGrabman in case you'll be working R package features

### CODE REVIEW

File: R-packages/covidcast/R/covidcast.R [hunk 1]
```diff
@@ -177,7 +177,9 @@ covidcast_signal <- function(data_source, signal,
                   .data$geo_type == given_geo_type)
 
   if (nrow(relevant_meta) == 0) {
-    relevant_meta <- list()
+    # even if no other metadata is available, we should set the geo_type so
+    # plotting functions can deal with this signal.
+    relevant_meta <- list(geo_type = geo_type)
```
Comment (sgsmob) [inline R-packages/covidcast/R/covidcast.R:182]: How much does this change the previous behavior?  Is it needed for testing?  All else equal, I would prefer that this PR only add tests.

**Review by sgsmob — APPROVED** [review]

File: R-packages/covidcast/tests/testthat/test-covidcast.R [hunk 2]
```diff
@@ -1,25 +1,42 @@
...
+# Many of these tests use mockery::with_mock_api. This replaces calls to the
+# live API server, instead returning static JSON files from disk. Three
+# weaknesses to be aware of:
+#
+# 1. The HTTP URL and parameters are hashed into a single filename to use. This
+# means that you can't have the query return one thing, then have the same query
+# return something else, for instance to test how the client handles different
+# types of errors.
+#
+# 2. Once you've written a test, it can be difficult to find the file storing
```
Comment (sgsmob) [inline R-packages/covidcast/tests/testthat/test-covidcast.R:14]: Can you also add comments in the json files to indicate to which query they belong?

Comment (capnrefsmmat) [reply R-packages/covidcast/tests/testthat/test-covidcast.R:14]: what's your preferred way for adding comments in JSON files? a dummy key/value pair?

File: R-packages/covidcast/R/covidcast.R (hunk 1 above)
Comment (capnrefsmmat) [reply R-packages/covidcast/R/covidcast.R:182]: See the commit message in 85c1ca2. It addresses the last point in #239. This behavior has regressed several times, and regressed again in #156.

I could separate the fix into another PR, but basically I got halfway through writing the test when I realized the code had regressed again, so I decided to include the fix. Separating it out may be more trouble than it's worth for a one-liner.

File: R-packages/covidcast/tests/testthat/test-covidcast.R (hunk 2 above)
Comment (sgsmob) [reply R-packages/covidcast/tests/testthat/test-covidcast.R:14]: Ah didn't realize json didn't support comments.  Ignore it then.

File: R-packages/covidcast/R/covidcast.R (hunk 1 above)
Comment (sgsmob) [reply R-packages/covidcast/R/covidcast.R:182]: Sounds good to leave it in as is

**Comment by capnrefsmmat** [comment]
@sgsmob so given the comments above, is this ready to merge?

**Comment by sgsmob** [comment]
yes

**Pull request merged**